### PR TITLE
Migration to webpack

### PR DESCRIPTION
**Pull request by 95kach**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,11 +4,13 @@
   "description": "Capsula JS - web components",
   "scripts": {
     "bootstrap": "lerna bootstrap --npm-client=yarn",
-    "build": "lerna run build",
-    "start": "lerna run start",
-    "lint": "lerna run lint",
-    "test": "lerna run test --concurrency=1",
-    "storybook": "start-storybook -p 6006"
+    "build": "lerna run build && node ./afterBuild.js --stream",
```
Comment (95kach) [inline package.json:7]: --stream - it's for logging

File: .storybook/tsconfig.json [hunk 2]
```diff
@@ -1,19 +1,8 @@
...
     "outDir": "./lib",
-    "moduleResolution": "node",
-    "baseUrl": "./src",
-    "esModuleInterop": true,
-    "jsx": "react",
-    "strict": true,
-    "noImplicitAny": true,
-    "noUnusedLocals": true,
-    "noUnusedParameters": true
+    "target": "es2015"
```
Comment (dmitriyslaym) [inline .storybook/tsconfig.json]: You have "target": "es2015" as a basic - I think it's redundant to specify it here

File: .storybook/tsconfig.json [hunk 3]
```diff
@@ -1,19 +1,8 @@
...
   },
-  "include": ["src/**/*"],
-  "exclude": ["node_modules", "**/*.spec.*"]
+  "include": ["./src/"]
```
Comment (dmitriyslaym) [inline .storybook/tsconfig.json:3]: you can extract it to the root tsconfig

File: .storybook/webpack.config.js [hunk 4]
```diff
@@ -1,14 +1,8 @@
...
-const include = path.resolve(__dirname, '../');
 
 module.exports = {
   module: {
     rules: [
-      {
-        test: /\.(js|jsx)$/,
-        exclude: /node_modules/,
-        loader: 'babel-loader',
-      },
```
Comment (dmitriyslaym) [inline .storybook/webpack.config.js:11]: why don't you need babel anymore?

File: package.json [hunk 5]
```diff
@@ -29,23 +31,37 @@
...
-    "babel-preset-react": "^6.24.1",
+    "@storybook/addon-actions": "^5.0.11",
+    "@storybook/addons": "^5.0.11",
+    "@storybook/react": "^5.0.11",
+    "@types/jest": "^24.0.13",
+    "@types/node": "^11.13.6",
+    "@types/react": "^16.8.14",
+    "@types/react-dom": "^16.8.4",
+    "@types/storybook__react": "^4.0.1",
+    "babel-polyfill": "^6.26.0",
```
Comment (dmitriyslaym) [inline package.json:42]: Are you sure that you need it as a dev dependency? It it is imported in the functionality files - so it's production dependency

File: packages/canvas/package.json [hunk 6]
```diff
@@ -12,40 +12,21 @@
...
-    "test": "npm-run-all -p -r start cypress"
+    "test": "npm-run-all -p -r start-example-server cypress"
   },
   "author": "Markets Pulse",
   "dependencies": {
-    "babel-polyfill": "^6.26.0",
-    "react": "^16.0.0",
-    "react-dom": "^16.8.6",
-    "@capsulajs-web-components/utils": "^0.0.1",
+    "@capsulajs/capsulahub-extension-utils": "^0.0.1",
```
Comment (dmitriyslaym) [inline packages/canvas/package.json:29]: I suggest to use the version "latest" for our internal packages

File: packages/canvas/src/Canvas.tsx [hunk 7]
```diff
@@ -13,7 +13,7 @@ export class Canvas extends HTMLElement {
 
   public connectedCallback() {
     const Component: React.JSXElementConstructor<any> = this.props$
-      ? dataComponentHoc(CanvasUI, this.props$)
+      ? dataComponentHoc(CanvasUI, this.props$ as any)
```
Comment (dmitriyslaym) [inline packages/canvas/src/Canvas.tsx]: why do you need as any here?

File: packages/canvas/tsconfig.json [hunk 8]
```diff
@@ -1,19 +1,8 @@
...
     "outDir": "./lib",
-    "moduleResolution": "node",
-    "baseUrl": "./src",
-    "esModuleInterop": true,
-    "jsx": "react",
-    "strict": true,
-    "noImplicitAny": true,
-    "noUnusedLocals": true,
-    "noUnusedParameters": true
+    "target": "es2015"
```
Comment (dmitriyslaym) [inline packages/canvas/tsconfig.json]: the same as in the root

File: packages/canvas/tsconfig.json [hunk 9]
```diff
@@ -1,19 +1,8 @@
...
   },
-  "include": ["src/**/*"],
-  "exclude": ["node_modules", "**/*.spec.*"]
+  "include": ["./src/"]
```
Comment (dmitriyslaym) [inline packages/canvas/tsconfig.json:6]: also can be in root

File: packages/configuration/package.json [hunk 10]
```diff
@@ -3,51 +3,26 @@
   "version": "0.0.1",
   "description": "Configuration service extension for CapsulaHub",
   "main": "lib/index.js",
```
Comment (dmitriyslaym) [inline packages/configuration/package.json:5]: I think the main should be dist

File: packages/configuration/tsconfig.json [hunk 11]
```diff
@@ -1,16 +1,8 @@
...
-    "noUnusedLocals": true,
-    "noUnusedParameters": true,
-    "moduleResolution": "node",
-    "baseUrl": "./src",
-    "esModuleInterop": true,
-    "jsx": "react"
-  }
+    "target": "es2015"
+  },
+  "include": ["./src/"]
```
Comment (dmitriyslaym) [inline packages/configuration/tsconfig.json:6]: exclude **target** and **include**

File: packages/logger/src/Logger.tsx [hunk 12]
```diff
@@ -13,7 +13,7 @@ export class Logger extends HTMLElement {
 
   public connectedCallback() {
     const Component: React.JSXElementConstructor<any> = this.props$
-      ? dataComponentHoc(LoggerUI, this.props$)
+      ? dataComponentHoc(LoggerUI, this.props$ as any)
```
Comment (dmitriyslaym) [inline packages/logger/src/Logger.tsx]: as any?

File: packages/logger/tsconfig.json [hunk 13]
```diff
@@ -1,19 +1,8 @@
...
-    "jsx": "react",
-    "strict": true,
-    "noImplicitAny": true,
-    "noUnusedLocals": true,
-    "noUnusedParameters": true
+    "target": "es2015"
   },
-  "include": ["src/**/*"],
-  "exclude": ["node_modules", "**/*.spec.*"]
+  "include": ["./src/"]
```
Comment (dmitriyslaym) [inline packages/logger/tsconfig.json:6]: exclude target and include

File: packages/request-form/tsconfig.json [hunk 14]
```diff
@@ -1,19 +1,8 @@
...
-    "jsx": "react",
-    "strict": true,
-    "noImplicitAny": true,
-    "noUnusedLocals": true,
-    "noUnusedParameters": true
+    "target": "es2015"
   },
-  "include": ["src/**/*"],
-  "exclude": ["node_modules", "**/*.spec.*"]
+  "include": ["./src/"]
```
Comment (dmitriyslaym) [inline packages/request-form/tsconfig.json:6]: target and include

File: packages/utils/tsconfig.json [hunk 15]
```diff
@@ -1,18 +1,8 @@
...
-    "jsx": "react",
-    "types": ["node", "react", "react-dom"],
-    "baseUrl": "./src",
-    "esModuleInterop": true,
-    "noImplicitAny": false
+    "target": "es2015"
   },
-  "include": ["src/**/*"],
-  "exclude": ["node_modules", "**/*.spec.*"]
+  "include": ["./src/"]
```
Comment (dmitriyslaym) [inline packages/utils/tsconfig.json:6]: target and include

File: .storybook/webpack.config.js (hunk 4 above)
Comment (95kach) [reply .storybook/webpack.config.js:11]: because now `canvas.story.js -> canvas.story.ts`

File: packages/canvas/src/Canvas.tsx (hunk 7 above)
Comment (95kach) [reply packages/canvas/src/Canvas.tsx]: because without it you will get Observable<CanvasProps> not match Observable<any>

**Review by dmitriyslaym — APPROVED** [review]

**Pull request merged**